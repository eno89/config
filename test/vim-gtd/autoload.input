" Tests for vim-gtd / autoload
function s:test_gtd_initialize_1()  "{{{1
  Title "gtd#initialize() - without any content"

  % delete _
  Assert getline(1, '$') ==> ['']

  call gtd#initialize()
  Assert getline(1, '$') ==> [
  \ 'INBOX',
  \ 'NEXT ACTIONS',
  \ 'CALENDAR',
  \ 'PROJECTS',
  \ 'WAITING FOR',
  \ 'SOMEDAY',
  \ 'ARCHIVE',
  \ 'TRASH',
  \ '; vim: filetype=gtd']
endfunction




function s:test_gtd_initialize_2()  "{{{1
  Title "gtd#initialize() - with some content"

  % delete _
  put =['a', 'b', 'c']
  Assert getline(1, '$') ==> ['', 'a', 'b', 'c']

  call gtd#initialize()
  Assert getline(1, '$') ==> [
  \ 'INBOX',
  \ 'NEXT ACTIONS',
  \ 'CALENDAR',
  \ 'PROJECTS',
  \ 'WAITING FOR',
  \ 'SOMEDAY',
  \ 'ARCHIVE',
  \ 'TRASH',
  \ '; vim: filetype=gtd']
endfunction




function s:test_gtd_status_1()  "{{{1
  Title "gtd#status() - without any content"

  % delete _
  Assert getline(1, '$') ==> ['']
  Assert gtd#status() ==> [0, 0]
endfunction




function s:test_gtd_status_2()  "{{{1
  Title "gtd#status() - with some content"

  % delete _
  put =['#1', 'x', '#3', '#999']
  Assert getline(1, '$') ==> ['', '#1', 'x', '#3', '#999']

  Assert gtd#status() ==> [999, 3]
endfunction




Test  "{{{1

" __END__
" vim: filetype=vim foldmethod=marker
