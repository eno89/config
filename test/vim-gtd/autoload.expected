===== "gtd#initialize() - without any content"
TEST: getline(1, '$')  ==>  [''] ... ok
TEST: getline(1, '$')  ==>  [ 'INBOX', 'NEXT ACTIONS', 'CALENDAR', 'PROJECTS', 'WAITING FOR', 'SOMEDAY', 'ARCHIVE', 'TRASH', '; vim: filetype=gtd'] ... ok
Result: 2 / 2

===== "gtd#initialize() - with some content"
TEST: getline(1, '$')  ==>  ['', 'a', 'b', 'c'] ... ok
TEST: getline(1, '$')  ==>  [ 'INBOX', 'NEXT ACTIONS', 'CALENDAR', 'PROJECTS', 'WAITING FOR', 'SOMEDAY', 'ARCHIVE', 'TRASH', '; vim: filetype=gtd'] ... ok
Result: 2 / 2

===== "gtd#status() - without any content"
TEST: getline(1, '$')  ==>  [''] ... ok
E486: Pattern not found: ^#
TEST: gtd#status()  ==>  [0, 0] ... ok
Result: 2 / 2

===== "gtd#status() - with some content"
4 more lines
TEST: getline(1, '$')  ==>  ['', '#1', 'x', '#3', '#999'] ... ok
TEST: gtd#status()  ==>  [999, 3] ... ok
Result: 2 / 2

===== Total
Result: 8 / 8
